### PR TITLE
feat(boostrap): added support for newer provider versions and improved bootstrap process 

### PR DESCRIPTION
**Pull request by SanderBlom**
## Description

- This upgrades the azurerm(v3 to v4), azuread(v2 to v3) and github (v5 to v6) providers to the newest version
- Fixes the issue with having to provide the subscription_id when running the bootstrap script https://github.com/blinqas/station/issues/173 
- Changed the name of the stations EntraID application and associated sp. Users also now have to option to specify a custom name.
- Changed some of the TF variables to have a default value. This should reduce the amount of variables users have to edit in the bootstrap script.
- Updated the docs with that permission you need to bootstrap station

## Type of Change

- [x] Bug fix
- [x] New feature
- [x] Documentation update
- [ ] Other

## Testing
See the testing [docs](../tests/readme.md) for more information about how to test your code 
- [ ] Performed a local `terraform plan`, `apply`, and `destroy` to validate changes.
- [ ] All tests are passing.
- [ ] Added, updated, or removed tests when appropriate

## Checklist

Before submitting this PR, please ensure the following:

- [x] My commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style guide
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings. I have ran `terraform validate` in the root module and all sub modules where I have made changes

## Additional Information

Provide any additional information or context about the pull request here.

---

Thank you for contributing to Station!

### CODE REVIEW

File: bootstrap/variables.tf [hunk 1]
```diff
@@ -81,4 +84,11 @@ variable "subscription_ids" {
   type        = set(string)
   description = "Set of Subscription ID's the Station identity can manage."
   default     = []
+}
+
+variable "station_entraID_application_name" {
```
Comment (kimfy) [inline bootstrap/variables.tf]: Nitpick: er `station_` nødvendig å inkludere i navnet?

File: bootstrap/bootstrap.sh [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
+export GITHUB_TOKEN=$TF_VAR_vcs_repo_PAT     # Authentication token for GitHub provider.
+export TF_VAR_vcs_repo_owner=$GITHUB_OWNER   # Set repository owner from above variable.
 
+# Azure-specific variables
+export ARM_SUBSCRIPTION_ID="Subscription_1"  # Azure Subscription ID for the provider. Here you can use the subscription_1 from TF_VAR_subscription_ids.
+
+# Optional variables that can be used to customize resource names
+#export TF_VAR_station_entraID_application_name = "station-deployments"     # Application name for the Entrada ID application that will create the new workloads in Azure.
+#export TF_VAR_vcs_repo_name="station-deployments"                          # Overrides the default Github Repository Name.
+#export TF_VAR_tfc_project_name="station"                                   # Project name in Terraform Cloud for 'station'.
```
Comment (kimfy) [inline bootstrap/bootstrap.sh]: Siden denne er kommentert ut vil ikke output på linje 68 ha en verdi å printe.

File: bootstrap/bootstrap.sh [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
+#export TF_VAR_deployments_tfc_workspace_name="station-deployments"         # TFC Workspace for station deployments.
```
Comment (kimfy) [inline bootstrap/bootstrap.sh]: Siden denne er kommentert ut vil ikke output på linje 69 ha en verdi å printe.

File: bootstrap/bootstrap.sh [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
+export GITHUB_OWNER="Github-username-or-org" # Target GitHub username or organization for the provider.
+export GITHUB_TOKEN=$TF_VAR_vcs_repo_PAT     # Authentication token for GitHub provider.
+export TF_VAR_vcs_repo_owner=$GITHUB_OWNER   # Set repository owner from above variable.
 
+# Azure-specific variables
+export ARM_SUBSCRIPTION_ID="Subscription_1"  # Azure Subscription ID for the provider. Here you can use the subscription_1 from TF_VAR_subscription_ids.
+
+# Optional variables that can be used to customize resource names
+#export TF_VAR_station_entraID_application_name = "station-deployments"     # Application name for the Entrada ID application that will create the new workloads in Azure.
+#export TF_VAR_vcs_repo_name="station-deployments"                          # Overrides the default Github Repository Name.
```
Comment (kimfy) [inline bootstrap/bootstrap.sh]: Siden denne er kommentert ut vil ikke output på linje 77 ha en verdi å printe.

**Review by kimfy — COMMENTED** [review]
Se kommentarene mine

**Review by emilhod — APPROVED** [review]

**Pull request merged**